### ISSUE TITLE
Reconcile Polaris tokens for v12

### ISSUE THREAD

**Issue by sam-b-rose** [Shopify/polaris-react-archive#9894]
TL;DR: Experimental Polaris design tokens need to be refined, merged into the token system, and documented so that 
consuming apps can migrate as part of the v12 upgrade path.

## Why

During the Polaris uplift build, a set of experimental design tokens were created to allow for iteration and to fill the gaps between the old and new design languages. These tokens need to be reconciled with the current token system now that the uplift is live and the design has stabilized. 

We will use this opportunity to reflect on ways the current token architecture and naming can make general usage and future design uplift a better experience for admin builders. The Polaris Migrator will be leveraged to enable a smooth upgrade path to Polaris v12. 

## Priorities

The current token system and the experimental uplift tokens will be merged:

- Unused and outdated tokens will be removed
- Naming will be aligned to scales, which may also need adjusting (e.g., 6px space token)
- Semantic layers will be added to ensure correct usage (e.g., z-index layers)
- Migrations will be written, documented, and added to the v12 migration steps

Theming of color will be considered as we’re updating the token system. In the future, we will roll out high-contrast themes and enable merchants to customize the admin. We want to ship admin design updates on a regular cadence that keeps Shopify feeling like new software.

## Guiding principles

- Tokens should be discoverable and exhaustive
- Token scales should have semantic aliases, where sensible, for consistent usage
- Theming consideration should be scoped to color only
- Theming should only be enabled at the app level—overrides are out of scope
- No breaking token changes will ship for at least 6 months after v12 is released

```[tasklist]
### Tasks
- [ ] https://github.com/Shopify/polaris/issues/9986
```

**Comment by alex-page** [comment]
Is this done?

**Comment by sam-b-rose** [comment]
Looking at the high-level description, I would consider this work complete.

@aaronccasanova, I see there are still a few unchecked items in the linked epic issue. Do you think we are good to close the epic issue (#1196) and this one?
